cypress/integration/interface_builder_test.js

```diff
@@ -1,3 +1,329 @@
+const _ = require('lodash');
+
+const parseInterfaceWithoutDefaults = ({ ...iface }) => {
+  if (iface.type === 'datastream' && iface.aggregation === 'individual') {
+    delete iface.aggregation;
+  }
+  if (iface.type === 'datastream') {
+    iface.mappings.forEach((mapping) => {
+      if (mapping.explicit_timestamp === false) {
+        delete mapping.explicit_timestamp;
+      }
+      if (mapping.reliability === 'unreliable') {
+        delete mapping.reliability;
+      }
+      if (mapping.retention === 'discard') {
+        delete mapping.retention;
+      }
+      if (mapping.expiry === 0) {
+        delete mapping.expiry;
+      }
+      if (mapping.database_retention_policy === 0) {
+        delete mapping.database_retention_policy;
+      }
+      if (mapping.database_retention_ttl === 60) {
+        delete mapping.database_retention_ttl;
+      }
+    });
+  }
+  if (iface.type === 'properties') {
+    iface.mappings.forEach((mapping) => {
+      if (mapping.allow_unset === false) {
+        delete mapping.allow_unset;
+      }
+    });
+  }
+  return iface;
+};
+
+const parseMappingOptions = (mapping) => {
+  return {
+    reliability: _.get(mapping, 'reliability', 'unreliable'),
+    explicitTimestamp: _.get(mapping, 'explicit_timestamp', false),
+    retention: _.get(mapping, 'retention', 'discard'),
+    expiry: _.get(mapping, 'expiry', 0),
+    databaseRetention: _.get(mapping, 'database_retention_policy', 'no_ttl'),
+    databaseTTL: _.get(mapping, 'database_retention_ttl', 60),
+    allowUnset: _.get(mapping, 'allow_unset', false),
+  };
+};
+
+const setupInterfaceEditorFromSource = (iface) => {
+  cy.get('#interfaceSource')
+    .clear()
+    .type(JSON.stringify(iface), { parseSpecialCharSequences: false });
```

Comment:
Definitely not. Here we want to know the least details possible about the implementation.

Reply:
but that's actually what you're testing, the internal representation of an interface.
In this code you're actually re-implementing the interface to json method, with no guarantees that the code will be bug-free.
If you don't want to use an internal method because we didn't unit test it. We should provide a valid interface and expect a specific outcome using fixtures, not calculating one ourself with the risk of testing the wrong implementation

Reply:
I'll rephrase. In this file we're not handling any interface JS implementation details of our app, we're only handling JSON representations (aka fixtures) of interfaces. We're not testing the internal workings of our app, the approach here is `black box` testing on the end result of the user actions.